packages/blockchain-wallet-v4-frontend/src/modals/ShowUsedAddresses/index.js

```diff
@@ -4,28 +4,34 @@ import { bindActionCreators, compose } from 'redux'
 
 import { actions } from 'data'
 import modalEnhancer from 'providers/ModalEnhancer'
-import ShowUsedAddresses from './template.js'
+import ShowUsedAddresses from './template'
 
 class ShowUsedAddressesContainer extends React.PureComponent {
   constructor (props) {
     super(props)
+    this.state = { busy: false }
     this.handleContinue = this.handleContinue.bind(this)
   }
 
   handleContinue () {
-    this.props.actions.toggleUsedAddresses(this.props.walletIndex, true)
+    this.setState({ busy: true })
+    // ensure busy is set before address derivation begins
+    setTimeout(() => {
+      this.props.actions.toggleUsedAddresses(this.props.walletIndex, true)
```

Comment:
why do you have timeout here? especially with 0?

Reply:
the derivation function is really slow when a wallets receiveIndex is > 15.  setTimeout forces the call to the function to the bottom of the event queue to ensure that the components state is set to busy, and thus a loading icon is presented to user, before the derivation process begins.